provider-example/src/kx.rs

```diff
@@ -43,19 +28,21 @@ impl crypto::KeyExchange for KeyExchange {
     fn group(&self) -> rustls::NamedGroup {
         X25519.name()
     }
-
-    fn all_kx_groups() -> &'static [&'static Self::SupportedGroup] {
-        ALL_KX_GROUPS
-    }
 }
 
 #[derive(Debug)]
 pub struct X25519;
 
-impl crypto::SupportedGroup for X25519 {
+impl crypto::SupportedKxGroup for X25519 {
     fn name(&self) -> rustls::NamedGroup {
         rustls::NamedGroup::X25519
     }
+
+    fn start(&self) -> Result<Box<dyn crypto::ActiveKeyExchange>, rustls::crypto::GetRandomFailed> {
```

Comment:
Nit: match the method order from the trait definition?

Also I'd probably inline the `priv_key` and `pub_key` bindings.

Reply:
reordered

have inlined `pub_key`, but `priv_key` needs to be a separate binding so `pub_key` can be derived from it